app/controllers/v1/courses_controller.rb

```diff
@@ -34,6 +35,10 @@ def course
       @course ||= Course.find(course_students_params[:course_id])
     end
 
+    def enrolment_exist
+      Enrolment.exists?(course: course, student: @current_user)
```

Comment:
This doesn't work because you'll return a list of courses, not a single one and therefore you don't have the course_id

Reply:
Create a method in the course model